ext/mbstring/tests/mb_convert_kana.phpt

```diff
@@ -16,27 +16,79 @@ $hanKakuB    =	'ｰｱｲｳｴｵｶｷｸｹｺｻｼｽｾｿ';
+echo bin2hex(mb_convert_kana("\xFF\x8A\xFF\x9F", 'HV', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\xFF\x8A\x00A", 'HV', 'UTF-16BE')), "\n";
+echo "Not using 'glue' mode:\n";
+echo bin2hex(mb_convert_kana("\xFF\x70", 'K', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\xFF\x70", 'H', 'UTF-16BE')), "\n\n";
+
+echo "Convert selected punctuation/symbols to full-width and back:\n";
+echo bin2hex(mb_convert_kana("\x00\x5C\x00\xA5\x00\x7E\x20\x3E\x00\x27\x00\x22", 'M', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\xFF\xE5\xFF\x3C\xFF\xE3\x20\x3E\x20\x18\x20\x19\x20\x1C\x20\x1D", 'm', 'UTF-16BE')), "\n\n";
+
+echo "Convert various full-width characters to half-width:\n";
+echo bin2hex(mb_convert_kana("\xFF\x01", 'a', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\xFF\x21", 'r', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\xFF\x10", 'n', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\x00", 's', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x22\x12", 'a', 'UTF-16BE')), "\n\n";
+
+echo "Convert full-width kana to half-width:\n";
+echo bin2hex(mb_convert_kana("\x30\x41", 'h', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\x01", 'h', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\x02", 'h', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\x0C", 'h', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\x0D", 'h', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\x9B", 'h', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\x9C", 'h', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\xFC", 'h', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\xFB", 'h', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\x01", 'k', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\x02", 'k', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\x0C", 'k', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\x0D", 'k', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\x9B", 'k', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\x9C", 'k', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\xFC", 'k', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\xFB", 'k', 'UTF-16BE')), "\n";
+echo "Including one which will expand to two codepoints:\n";
+echo bin2hex(mb_convert_kana("\x30\x52", 'h', 'UTF-16BE')), "\n\n";
+
+echo "Convert full-width hiragana to full-width katakana:\n";
+echo bin2hex(mb_convert_kana("\x30\x41", 'C', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\x9D", 'C', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\x9E", 'C', 'UTF-16BE')), "\n\n";
+
+echo "Convert full-width katakana to full-width hiragana:\n";
+echo bin2hex(mb_convert_kana("\x30\xA1", 'c', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\xFD", 'c', 'UTF-16BE')), "\n";
+echo bin2hex(mb_convert_kana("\x30\xFE", 'c', 'UTF-16BE')), "\n\n";
+
+echo bin2hex(mb_convert_kana("\x00\x00", 'A', 'UTF-16BE')), "\n";
```

Comment:
Any reason the new tests work on UTF-16BE and bin2hex, rather than UTF-8?

Reply:
Because with UTF-16, which codepoints are being tested (and what codepoints they are converted to) can easily be seen. For example, right there you're seeing a test involving `U+30FD`, another involving `U+30FE`, and so on. In UTF-8, the hex form of the raw bytes don't correspond trivially to the codepoint number.

Reply:
I was mainly thinking of just using non-encoded characters here, like ヽ and ヾ.

Reply:
Hmm. I don't like that idea much for `U+3000`.

But if you feel strongly about the others, I can convert them. Personally, seeing the raw values in hex format is good for me, since that is what appears in the corresponding C code. Maybe it wouldn't be as good for a casual reader.

Up to you.